### PR TITLE
GB-11 - Create tests

### PR DESCRIPTION
**Pull request by veranika107**

### CODE REVIEW

**Comment by veranika107** [comment]
[GB - Add more tests](https://trello.com/c/qxTKtuYx/11-gb-add-more-tests)

File: tests/Controller/ConferenceControllerTest.php [hunk 1]
```diff
@@ -33,7 +45,7 @@ public function testConferencePage()
         $this->assertSelectorExists('div:contains("There is one comment")');
     }
 
-    public function testCommentSubmission()
+    public function testCommentSubmission(): void
```
Comment (wechka53) [inline tests/Controller/ConferenceControllerTest.php:48]: conference controller -> method show has this block
try {
                    $photo->move($photoDir, $filename);
                } catch (FileException $e) {
                    print_r($e->getMessage());
                }
the test should cover this case as well

**Review by wechka53 — CHANGES_REQUESTED** [review]
Missing tests for commands + few small improvements required

File: config/services.yaml [hunk 2]
```diff
@@ -26,5 +26,21 @@ services:
...
+            autowire: true      # Automatically injects dependencies in your services.
+            autoconfigure: true # Automatically registers your services as commands, event subscribers, etc.
+
+        App\:
+            resource: '../src/'
+            exclude:
+                - '../src/DependencyInjection/'
+                - '../src/Entity/'
+                - '../src/Kernel.php'
+        App\Service\SpamChecker:
```
[the remark this answers is not in the document]
Comment (veranika107) [reply config/services.yaml:41]: As I know all services are private. Only in version 6.3 there will be no need to declare services as private for testing need https://symfony.com/doc/current/testing.html#mocking-dependencies

File: tests/Service/ImageOptimizerTest.php
[the remark this answers is not in the document]
Comment (veranika107) [reply tests/Service/ImageOptimizerTest.php:1]: There is a test that check initial image ratio and the ratio of resized image, I think it covers this case

[the remark this answers is not in the document]
Comment (wechka53) [reply tests/Service/ImageOptimizerTest.php:1]: it doesn't 
image
You can run phpunit with coverage to check it by yourself

File: config/services.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (wechka53) [reply config/services.yaml:41]: Indeed

File: tests/MessageHandler/CommentMessageHandlerTest.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        $this->commentRepository->expects($this->once())
+            ->method('find')
+            ->willReturn(null);
+        $this->container->set(CommentRepository::class, $this->commentRepository);
+
+        /** @var CommentMessageHandler $commentMessageHandler */
+        $commentMessageHandler = $this->container->get(CommentMessageHandler::class);
+
+        try {
+            $commentMessageHandler($this->commentMessage);
```
[the remark this answers is not in the document]
Comment (veranika107) [reply tests/MessageHandler/CommentMessageHandlerTest.php:53]: there are no thrown exceptions in the code. There will be an error only if someone will remove "if" condition from the code. And in the test I want to catch this error and fail the test with described exception

[the remark this answers is not in the document]
Comment (wechka53) [reply tests/MessageHandler/CommentMessageHandlerTest.php:53]: We could try to do more correct way to test this case. For instance:
 1   if the comment is null, message handler will return void or null. 
 2.  $this->commentStateMachine->can() never be called, right?

File: tests/Controller/ConferenceControllerTest.php (hunk 1 above)
Comment (veranika107) [reply tests/Controller/ConferenceControllerTest.php:48]: Thank you for noticing, fixed

File: tests/Service/ImageOptimizerTest.php
[the remark this answers is not in the document]
Comment (veranika107) [reply tests/Service/ImageOptimizerTest.php:1]: Thanks, I checked coverage. Now it is fully covered

**Review by wechka53 — APPROVED** [review]

**Pull request merged**